src/test/java/com/selfxdsd/selfweb/LoginProducerTestCase.java

```diff
@@ -0,0 +1,111 @@
+package com.selfxdsd.selfweb;
+
+import com.selfxdsd.api.Login;
+import com.selfxdsd.api.Self;
+import com.selfxdsd.api.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+
+/**
+ * Unit tests for {@link LoginProducerTestCase}.
+ * @author Nikita Monokov (nmonokov@gmail.com)
+ * @version $Id$
+ * @since 0.0.1
+ */
+public final class LoginProducerTestCase {
+
+    /**
+     * Logins with Github provider.
+     */
+    @Test
+    public void githubLogin() {
+        final SecurityContext securityContext = Mockito.mock(SecurityContext.class);
+        final OAuth2AuthenticationToken authentication = Mockito.mock(OAuth2AuthenticationToken.class);
+        SecurityContextHolder.setContext(securityContext);
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+        Mockito.when(SecurityContextHolder.getContext().getAuthentication())
+                .thenReturn(authentication);
+        final OAuth2AuthenticationToken oauthTokenMock =
+                (OAuth2AuthenticationToken) SecurityContextHolder.getContext().getAuthentication();
+        final Self core = Mockito.mock(Self.class);
+        final User authenticated = Mockito.mock(User.class);
+        Mockito.when(core.login(Mockito.any()))
+                .thenReturn(authenticated);
+        Mockito.when(oauthTokenMock.getAuthorizedClientRegistrationId())
+                .thenReturn("github");
+        final OAuth2AuthorizedClientService clientService =
+                Mockito.mock(OAuth2AuthorizedClientService.class);
+
+        LoginProducer producer = new LoginProducer(core);
+        producer.login(clientService);
+
+        Mockito.verify(core, Mockito.only()).login(
+                Mockito.any(Login.class)
+        );
```

Comment:
@nmonokov Can you also add an assertion that the returned User is actually the ``authenticated`` User declared above?

Something like:

```java
MatcherAssert.assertThat(
    producer.login(clientService),
    Matchers.is(authenticated)
)
```

And the same for the test bellow.